nat/nat.tf

```diff
@@ -28,7 +28,7 @@ resource "aws_route" "main" {
 
 # Create route tables so elastic IPs are used for outgoing traffic
 resource "aws_route_table" "main" {
-  count  = length(var.subnet_public_ids)
+  count  = length(var.public_ips)
```

Comment:
we later iterate over (private) subnets and get the route table for each (private) subnet. Does the number of public IPs necessarily have to be the same as subnets? if yes, this change is pointless, if no, the route table association will fail since the count is different.